tests/pauli_rotations/generate_tests_circuit.py

```diff
@@ -0,0 +1,200 @@
+    c3.add_pauli_block(PauliRotation.from_list([I, Z, Z, I], Fraction(1, 8)))
+    c3.add_pauli_block(Measurement.from_list([Y, X, I, I]))
+    if case == "eq":
+        return [(c1, c1), (c2, c2), (c3, c3)]
+    elif case == "ne":
+        return [(c1, c2), (c2, c3), (c3, c1)]
+    else:
+        assert False
+
+
+def generate_tests_circuit_has_measurements() -> List[Tuple[bool, bool]]:
+    c1 = PauliOpCircuit(4)
+    c1.add_pauli_block(PauliRotation.from_list([Z, X, I, Y], Fraction(1, 8)))
+    c1.add_pauli_block(PauliRotation.from_list([Z, X, I, Y], Fraction(-1, 8)))
+
+    c2 = PauliOpCircuit(7)
+    c2.add_pauli_block(PauliRotation.from_list([X, Y, Z, I, X, Z, Y], Fraction(-1, 4)))
+    c2.add_pauli_block(PauliRotation.from_list([I, Z, Z, I, X, X, X], Fraction(1, 8)))
+    c2.add_pauli_block(Measurement.from_list([Z, X, I, Z, I, I, I]))
+    c2.add_pauli_block(Measurement.from_list([I, I, Z, I, X, X, Z]))
+
+    c3 = PauliOpCircuit(2)
+    c3.add_pauli_block(PauliRotation.from_list([Z, Z], Fraction(1, 4)))
+    c3.add_pauli_block(Measurement.from_list([X, X]))
+
+    return [
+        (c1.circuit_has_measurements(), False),
+        (c2.circuit_has_measurements(), True),
+        (c3.circuit_has_measurements(), True),
+    ]
+
+
+def generate_tests_are_commuting() -> List[Tuple[bool, bool]]:
+    tests_list = list()
+    block1 = PauliRotation.from_list([X, I, Z, I], Fraction(1, 4))
+    block2 = PauliRotation.from_list([I, Z, I, X], Fraction(-1, 4))
+    block3 = PauliRotation.from_list([Z, X, I, Z], Fraction(1, 8))
+    tests_list.append((PauliOpCircuit.are_commuting(block1, block1), True))
+    tests_list.append((PauliOpCircuit.are_commuting(block1, block2), True))
+    tests_list.append((PauliOpCircuit.are_commuting(block1, block3), False))
+    tests_list.append((PauliOpCircuit.are_commuting(block2, block3), True))
+
+    return tests_list
+
+
+def generate_tests_apply_transformation():
+    """
+    Three Litinski Rules
+    1. Test 1 --> P, P` commute - Litinski 4a
+        Note: Whenever two PauliRotation block are checked from commutation, they always
+        commute in the implemented test case
+    2. Test 2 --> P, P` anti commute - Litinski 4a
+        Note: Whenever two PauliRotation block are checked from commutation, they always
+        anti-commute in the implemented test case
+    3. Test 3 --> P, P' commute - Litinski 4b
+    4. Test 4 --> P,P' anti commute - Litinski 4b
+    5. Test 5 --> Controlled Operations, commute - Litinski 4c
+        Tests for this is implicitly included in tests 1 and 2, as controlled operations
+        are converted Pauli Rotations in load_*() methods
+    6. Test 6 --> Controlled Operations, anticommute - Litinski 4c
+        Tests for this is implicitly included in tests 1 and 2, as controlled operations
+        are converted Pauli Rotations in load_*() methods
+
+    `PauliOpCircuit.commute_pi_over_four_rotations()` are also implicitly tested through these tests
```

Comment:
Per https://github.com/latticesurgery-com/lattice-surgery-compiler/issues/156#issuecomment-973179680, I suggest that we add tests for `PauliOpCircuit.commute_pi_over_four_rotations()`, given that it is an important building block for other algorithms and is used more widely than `apply_transformation()`. Though, it should be another PR.

Reply:
Agreed. Let's merge this branch with the changes(after review) and then solve #156 on a seperate PR